### ISSUE TITLE
Add support for XML 1.1

### ISSUE THREAD

**Issue by napsta32** [mikefarah/yq#2047]
**Feature request.**

Currently, XML 1.1 is not supported. See this log when you try to import XML 1.1:

```
...xml: unsupported version "1.1"; only version 1.0 is supported
```

**Additional context**

I want to transform XML documents into JSON.
XML 1.1 is used by Jenkins plugins to pretty-format logs and log outputs.

**Comment by keithbowes** [comment]
Yeah, it seems that Jenkins is also the use case specified in the bug report about adding XML 1.1 to libxml2. I'd have to wonder why Jenkins decided to use XML 1.1 when nothing supports it; to support logs with control characters forbidden in XML 1.0 (the rationale they give for using XML 1.1), they should have sanitized the input for XML 1.0 instead of using XML 1.1. XML 1.1 was pretty much DOA when it was initially published. Anyway what I'd recommend would be something like `cat file.xml | sed -e 's/<?xml version="1.1"/<?xml version="1.0"/' | yq .` if you're not using the aforementioned control characters.
